data.go

```diff
@@ -87,8 +94,18 @@ func (r *dataReader) Read(b []byte) (n int, err error) {
 		stateEOF              // reached .\r\n end marker line
 	)
 	for n < len(b) && r.state != stateEOF {
+		if r.c.server.ReadTimeout != 0 {
+			err = r.c.conn.SetReadDeadline(time.Now().Add(r.c.server.ReadTimeout))
+			if err != nil {
+				break
+			}
+			if e, ok := err.(net.Error); ok && e.Timeout() {
```

Comment:
Hm, so this checks a `SetReadDeadline` error? Weird, I would've expected to check the `ReadByte` error instead?

BTW, a better way to check for the timeout is: `errors.Is(err, os.ErrDeadlineExceeded)` (see https://pkg.go.dev/net#Conn).